apps/api/src/schema/catalogue.ts

```diff
@@ -0,0 +1,105 @@
+import { z } from "@hono/zod-openapi";
+
+export const sampleProgramsQuerySchema = z.object({
+  id: z.string().optional().openapi({
+    description: "Filter sample programs by program ID and returns all programs if omitted.",
+    example: "computerscience_bs",
+  }),
+});
+
+export const standingyearEnum = z.enum(["Freshman", "Sophomore", "Junior", "Senior"]);
+
+export const courseEntrySchema = z.discriminatedUnion("type", [
+  z.object({
+    type: z.literal("courseId"),
+    value: z.string().openapi({
+      description: "Course ID that was validated and found in the database",
+      example: "I&CSCI31",
+    }),
+  }),
+  z.object({
+    type: z.literal("unknown"),
+    value: z.string().openapi({
+      description:
+        "Original text that could not be validated as a course (e.g., general education requirements)",
+      example: "General Education III",
+    }),
+  }),
+]);
+
+export const sampleProgramsYearSchema = z
+  .object({
+    year: standingyearEnum.openapi({
+      description: "Class standing or year level",
+    }),
+    fall: z.array(courseEntrySchema).openapi({
+      description:
+        "Courses recommended for Fall term. Each entry is either a validated course ID or descriptive text.",
+    }),
+    winter: z.array(courseEntrySchema).openapi({
+      description:
+        "Courses recommended for Winter term. Each entry is either a validated course ID or descriptive text.",
+    }),
+    spring: z.array(courseEntrySchema).openapi({
+      description:
+        "Courses recommended for Spring term. Each entry is either a validated course ID or descriptive text.",
+    }),
+  })
+  .openapi({
+    example: {
+      year: "Freshman",
+      fall: [
+        { type: "courseId", value: "I&CSCI31" },
+        { type: "courseId", value: "MATH2A" },
+        { type: "courseId", value: "WRITING40" },
+      ],
+      winter: [
+        { type: "courseId", value: "I&CSCI32" },
+        { type: "courseId", value: "MATH2B" },
+        { type: "courseId", value: "WRITING50" },
+        { type: "unknown", value: "General Education III" },
+      ],
+      spring: [
+        { type: "courseId", value: "I&CSCI33" },
+        { type: "courseId", value: "IN4MATX43" },
+        { type: "courseId", value: "I&CSCI6B" },
+        { type: "courseId", value: "WRITING60" },
+      ],
+    },
+  });
+
+export const sampleProgramVariationSchema = z.object({
+  label: z.string().nullable().openapi({
+    description:
+      "Label describing this variation of the multiple sample programs available for this program",
```

Comment:
I'm not sure we have to tie the existence of a label to the fact that there are multiple variations. I know that's actually the case, but each of these variations should be valid in its own right without concern for what's going on around it. For that reason, you should just describe this field without reference to other variations (including whether they exist or not).